ci-operator/step-registry/openshift-qe/loaded-upgrade/openshift-qe-loaded-upgrade-commands.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+cat /etc/os-release
+oc config view
+oc projects
+pushd /tmp
+python3 --version
+python3 -m venv venv3
+source venv3/bin/activate
+pip3 --version
+pip3 install --upgrade pip
+pip3 install -U datetime pyyaml
+pip3 list
+
+RELEASE_IMAGE_LATEST=${RELEASE_IMAGE_LATEST:=""}
+CURRENT_VERSION=$(oc get clusterversion -ojsonpath={..desired.version})
+OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE=${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE:=$RELEASE_IMAGE_LATEST}
+
+if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
+    echo "RELEASE_IMAGE_LATEST is an empty string, exiting"
+    exit 1
+fi
+
+if [[ -s "${SHARED_DIR}/perfscale-override-upgrade" ]]; then
+      ALL_IMAGES="$(< "${SHARED_DIR}/perfscale-override-upgrade")" &&
+      echo "Overriding upgrade target to ${ALL_IMAGES}"
+      for IMAGE in $ALL_IMAGES
+      do
+	      RELEASES_VERSION+=("`oc adm release info ${IMAGE} --output=json | jq -r '.metadata.version'`")
+      done
+      TARGET_RELEASES=$(echo "${RELEASES_VERSION[@]}"| tr -s ' ' ',')
+else
+      TARGET_RELEASES="$(oc adm release info "${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}" --output=json | jq -r '.metadata.version')"
+fi
+echo  "-------------------------------------------------------------------------------------------"
+echo  Loaded Upgrade from [ $CURRENT_VERSION ] to [ $TARGET_RELEASES ]
+echo  "-------------------------------------------------------------------------------------------"
+
+if [ -d ocp-qe-perfscale-ci ];then
+     rm -rf ocp-qe-perfscale-ci
+fi
+git clone -b upgrade https://github.com/liqcui/ocp-qe-perfscale-ci.git --depth=1
```

Comment:
Are there any changes that are in your fork of the main ocp-qe-perfscale repo that we need to add in to the main upgrade branch? Think we should add the main repo when merging: _https://github.com/openshift-qe/ocp-qe-perfscale-ci.git_

Reply:
No new change for this script, update it in new code, but should we make a copy of upgrade script to main branch? we use main branch script for prow ci in the future?

Reply:
Any branch in the https://github.com/openshift-qe/ocp-qe-perfscale-ci.git repo is considered the "main" branch for its functionality. That repo uses separate branches to define a different type of functionality for our testing so the **upgrade** branch on the openshift-qe based repo is final and has restricted merge access to be sure we can keep it in a stable state